code/modules/power/generator.dm

```diff
@@ -183,8 +222,15 @@
 		else
 			disconnect_from_network()
 		reconnect()
-	else
-		..()
+		return TRUE
+
+	if(default_deconstruction_screwdriver(user, attacking_item))
+		return TRUE
+	if(default_deconstruction_crowbar(user, attacking_item))
+		return TRUE
+	if(default_part_replacement(user, attacking_item))
+		return TRUE
+	return ..()
```

Comment:
This should be an elseif chain, and the last return with the call to parent should probably be the last `else` by the looks of it